text/0000-roadmap-2019.md

```diff
@@ -0,0 +1,330 @@
+- **Improved IDE integration**, including support for typed completions
+- Extracting parts of rustc into **libraries**, which are easier to understand and maintain but also help in developing a Rust specification
+- Supporting lang team in **key feature development**, such as around `async fn` support, or improvements to the trait system 
+- Improving our **documentation** and **organization**, helping to make it clearer what is going on at any given time, and how people can get involved.
+
+Some of the specific projects we have in mind are as follows:
+
+- **Parallelizing rustc**, which should help with improving raw compilation times and also increasing IDE responsiveness. 
+- Introducing **MIR optimizations**, which will improve the quality of our generated code, but also may help reduce compilation time by giving LLVM less work to do (LLVM remains a large percentage of overall compilation time).
+- Introducing **working groups** for each project, and track them on the [compiler-team repository](https://github.com/rust-lang/compiler-team).
+
+One project worth calling out in more detail is the [**RLS 2.0
+effort**](https://github.com/rust-lang/compiler-team/tree/master/working-groups/rls-2.0).
+The existing RLS, while functional, was never deeply integrated into rustc
+itself. Over the past two years, we have rewritten the back end of rustc to
+use an **incremental and demand-driven infrastructure.** To get the IDE
+experience we truly want, we need to see that work through for the front end
+of the compiler as well. However, this part of rustc is one of its oldest,
+and there are significant design questions involved in how best to do that.
+
+This is where the RLS 2.0 effort comes in. **The plan is to build a prototype
+of the new front-end, thus enabling a truly first-class IDE experience.**
+Wherever possible, we aim to share code between this front-end and rustc by
+extracting libraries, such as [Chalk] (for traits), [Polonius] (for the
+borrow checker), and a new library focusing on name resolution. Eventually,
+we will merge this new front-end with the remaining bits of back-end from
+rustc.
+
+[Chalk]: https://github.com/rust-lang-nursery/chalk
+[Polonius]: http://github.com/rust-lang-nursery/polonius
+[rust-analyzer project]: https://github.com/rust-analyzer/rust-analyzer
+
+## Crates.io
+
+One of the newer teams, the Crates.io Team is going to be focused on growing
+itself, paying down technical debt in the codebase, and establishing themes
+for long-term priorities.
+
+## Dev Tools
+
+The Dev Tools team has a few core functions:
+
+- liaising with the wider project
+- being a bit of a "commons" space for cross-cutting tools concerns
+- finding out what needs are yet to be addressed by tooling and filling them
+- providing some kind of quorum helping smaller tool subteams make decisions
+
+They'll be working to fulfill these functions this year, along with looking
```

Comment:
For a roadmap, the approach of listing "core functions" and then saying the team will work to fulfill these functions this year seems a bit weak. Is that different from last year, and if so, how? If all the specific measurable goals are pursued by subteams, maybe this section can just be left out?

Reply:
It is different from last year, last year the devtools team didn't really have these core functions. It somewhat implicitly did but didn't necessarily actively pursue them. We've only recently enumerated these and decided to start moving towards them.

I don't consider the roadmap to be a place for _only_ measurable goals.